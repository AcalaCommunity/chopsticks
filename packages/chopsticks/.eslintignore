vitest.config.ts
.eslintrc.js
node_modules/
dist/
lib/
preview/
chopsticks.js
