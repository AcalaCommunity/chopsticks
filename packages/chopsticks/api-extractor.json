{
	"extends": "../../api-extractor.base.json",
	"mainEntryPointFilePath": "./lib/types.d.ts"
}
