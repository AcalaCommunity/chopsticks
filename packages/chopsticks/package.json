{
	"name": "@acala-network/chopsticks",
	"version": "0.9.1-2",
	"author": "Acala Developers <hello@acala.network>",
	"license": "Apache-2.0",
	"bin": "./chopsticks.js",
	"scripts": {
		"clean": "rm -rf dist",
		"build": "yarn clean && tsc -p ./tsconfig.json && tsc -p ./tsconfig.esm.json && echo '{\"type\": \"module\"}' > ./dist/esm/package.json && yarn copyfiles",
		"copyfiles": "cp -r src/utils/template dist/cjs/utils/ && cp -r src/utils/template dist/esm/utils/",
		"script:start": "cd ../..; ts-node --transpile-only -r tsconfig-paths/register packages/chopsticks/src/cli.ts",
		"script:run": "cd ../..; LOG_LEVEL=trace ts-node-dev --transpile-only -r tsconfig-paths/register --inspect --notify=false packages/chopsticks/src/cli.ts -- --config=configs/dev.yml",
		"dev:karura": "cd ../..; ts-node-dev --transpile-only --inspect -r tsconfig-paths/register --notify=false packages/chopsticks/src/cli.ts -- --config=configs/karura.yml",
		"dev:acala": "cd ../..; ts-node-dev --transpile-only --inspect -r tsconfig-paths/register --notify=false packages/chopsticks/src/cli.ts -- --config=configs/acala.yml",
		"dev:polkadot": "cd ../..; ts-node-dev --transpile-only --inspect -r tsconfig-paths/register --notify=false packages/chopsticks/src/cli.ts -- --config=configs/polkadot.yml",
		"dev:moonriver": "cd ../..; ts-node-dev --transpile-only --inspect -r tsconfig-paths/register --notify=false packages/chopsticks/src/cli.ts -- --config=configs/moonriver.yml",
		"dev:moonbeam": "cd ../..; ts-node-dev --transpile-only --inspect -r tsconfig-paths/register --notify=false packages/chopsticks/src/cli.ts -- --config=configs/moonbeam.yml",
		"docs:prep": "typedoc"
	},
	"dependencies": {
		"@acala-network/chopsticks-core": "workspace:*",
		"@acala-network/chopsticks-db": "workspace:*",
		"@pnpm/npm-conf": "^2.2.2",
		"axios": "^1.6.0",
		"dotenv": "^16.3.1",
		"global-agent": "^3.0.0",
		"js-yaml": "^4.1.0",
		"jsondiffpatch": "^0.5.0",
		"lodash": "^4.17.21",
		"ws": "^8.14.2",
		"yargs": "^17.7.2",
		"zod": "^3.22.3"
	},
	"devDependencies": {
		"@types/global-agent": "^2.1.1",
		"@types/js-yaml": "^4.0.5",
		"@types/lodash": "^4.14.199",
		"@types/node": "^20.5.7",
		"@types/ws": "^8.5.6",
		"@types/yargs": "^17.0.29",
		"ts-node": "^10.9.1",
		"ts-node-dev": "^2.0.0",
		"typescript": "^5.1.6"
	},
	"files": [
		"dist/esm/**",
		"dist/cjs/**",
		"dist/types/**",
		"chopsticks.js"
	],
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"types": "./dist/types/index.d.ts",
	"exports": {
		".": {
			"types": "./dist/types/index.d.ts",
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./*": {
			"types": "./dist/types/index.d.ts",
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./plugins/*": {
			"types": "./dist/types/plugins/*.d.ts",
			"require": "./dist/cjs/plugins/*.js",
			"import": "./dist/esm/plugins/*.js",
			"default": "./dist/esm/plugins/*.js"
		},
		"./package.json": "./package.json",
		"./package.esm.json": "./dist/esm/package.json"
	}
}
