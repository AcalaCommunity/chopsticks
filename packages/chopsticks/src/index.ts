import '@polkadot/api-augment'

export * from '@acala-network/chopsticks-core'
export { setupWithServer } from './setup-with-server'
export { fetchConfig } from './schema'
