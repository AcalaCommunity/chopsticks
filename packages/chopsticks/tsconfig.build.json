{
	"extends": "../../tsconfig.base.json",
	"compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib",
    "rootDir": "./src"
  },
	"exclude": ["node_modules", "vendor", "executor", "e2e", "vitest.config.ts"]
}
