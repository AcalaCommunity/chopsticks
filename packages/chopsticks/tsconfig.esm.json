{
	"extends": "./tsconfig.json",
	"compilerOptions": {
		"module": "ES2022",
		"outDir": "dist/esm"
	},
	"references": [{ "path": "../core/tsconfig.esm.json" }, { "path": "../db/tsconfig.esm.json" }]
}
