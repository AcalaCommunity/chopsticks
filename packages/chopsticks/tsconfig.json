{
	"extends": "../../tsconfig.base.json",
	"compilerOptions": {
		"outDir": "lib",
		"rootDir": "src"
	},
	"include": ["src/**/*"],
	"exclude": ["src/**/*.test.ts"],
	"references": [{ "path": "../core" }, { "path": "../db" }],
	"typedocOptions": {
		"entryPoints": ["src/types.ts"],
		"out": "../../docs-src/chopsticks",
		"plugin": "typedoc-plugin-markdown",
		"readme": "none"
	}
}
