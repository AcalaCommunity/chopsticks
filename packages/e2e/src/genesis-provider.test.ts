import { describe, expect, it } from 'vitest'

import { BuildBlockMode } from '@acala-network/chopsticks'
import { check, testingPairs } from './helper.js'
import { setupContextWithConfig } from '@acala-network/chopsticks-testing'

describe.each([
  [
    'Mandala',
    'https://raw.githubusercontent.com/AcalaNetwork/Acala/2c43dbbb380136f2c35bd0db08b286f346b71d61/resources/mandala-dist.json',
  ],
  ['Kusama', new URL('../blobs/kusama.json', import.meta.url).pathname],
  ['Asset Hub Kusama', new URL('../blobs/asset-hub-kusama.json', import.meta.url).pathname],
])(`genesis provider works %s`, async (name, genesis) => {
  const { chain, dev, api } = await setupContextWithConfig({
    port: 1234,
    genesis,
    'build-block-mode': BuildBlockMode.Manual,
  })

  describe('genesis provider works', () => {
    it('build blocks', async () => {
      expect(await dev.newBlock()).toBeTruthy()
      const block = await chain.getBlockAt(1)
      expect(block).toBeTruthy
      expect(block?.number).toBe(1)
      await check(api.rpc.system.name()).toMatchSnapshot()
    })

    it.skipIf(name === 'Asset Hub Kusama')('handles tx', async () => {
      await dev.newBlock()

      const { alice, bob } = testingPairs()

      await dev.setStorage({
        System: {
          Account: [[[alice.address], { providers: 1, data: { free: 1000 * 1e12 } }]],
        },
      })

      await check(api.query.system.account(alice.address)).toMatchSnapshot()

      await api.tx.balances.transfer(bob.address, 123 * 1e12).signAndSend(alice)

      await dev.newBlock()

      await check(api.query.system.account(alice.address)).toMatchSnapshot()
      await check(api.query.system.account(bob.address)).toMatchSnapshot()
    })
  })
})
