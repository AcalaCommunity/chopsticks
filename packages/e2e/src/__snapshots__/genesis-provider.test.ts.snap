// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`genesis provider works Asset Hub Kusama > genesis provider works > build blocks 1`] = `"Statemine Chopsticks"`;

exports[`genesis provider works Kusama > genesis provider works > build blocks 1`] = `"Parity-kusama Chopsticks"`;

exports[`genesis provider works Kusama > genesis provider works > handles tx 1`] = `
{
  "consumers": 0,
  "data": {
    "flags": 0,
    "free": 1000000000000000,
    "frozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`genesis provider works Kusama > genesis provider works > handles tx 2`] = `
{
  "consumers": 0,
  "data": {
    "flags": "0x80000000000000000000000000000000",
    "free": 876999432127266,
    "frozen": 0,
    "reserved": 0,
  },
  "nonce": 1,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`genesis provider works Kusama > genesis provider works > handles tx 3`] = `
{
  "consumers": 0,
  "data": {
    "flags": "0x80000000000000000000000000000000",
    "free": 123000000000000,
    "frozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`genesis provider works Mandala > genesis provider works > build blocks 1`] = `"Mandala Chopsticks"`;

exports[`genesis provider works Mandala > genesis provider works > handles tx 1`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 1000000000000000,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`genesis provider works Mandala > genesis provider works > handles tx 2`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 876982732391839,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 1,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`genesis provider works Mandala > genesis provider works > handles tx 3`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 123000000000000,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;
